3_managing_roles/README.md

```diff
@@ -1,3 +1,108 @@
 # Managing Roles
 
-A collection of examples regarding managing roles.
+A collection of examples regarding managing Roles.
+
+## Migrating to Access Rules
+
+To increase flexibility when managing thousands of Resources, Role Grants have
+been deprecated in favor of Access Rules, which allow you to grant access based
+on Resource Tags and Type. Role Grants will be removed from the Terraform
+provider by March 15, 2022.
+
+The following examples demonstrate the deprecated Role Grants, Dynamic Access
+Rules with Tags and Resource Types, and Static Access Rules for backwards
+compatibility with Role Grants.
+
+### Role Grants (deprecated)
+
+Previously, you would grant a Role access to specific resources by ID via Role
+Grants:
+
+```hcl
+resource "sdm_resource" "redis-test" {
+  redis {
+    name = "redis-test"
+    hostname = "example.com"
+    port_override = 4020
+    tags = {
+      region = "us-west"
+      env = "dev"
+    }
+  }
+}
+
+resource "sdm_resource" "postgres-test" {
+  postgres {
+    name = "postgres-test"
+    hostname = "example.com"
+    database = "my-db"
+    username = "admin"
+    password = "hunter2"
+    port = 5432
+    tags = {
+      region = "us-west"
+      env = "dev"
+    }
+  }
+}
+
+resource "sdm_role" "engineering" {
+  name = "engineering"
+}
+
+resource "sdm_role_grant" "engineering-redis" {
+  resource_id = sdm_resource.redis-test.id
+  role_id = sdm_role.engineering.id
+}
+
+resource "sdm_role_grant" "engineering-postgres" {
+  resource_id = sdm_resource.postgres-test.id
+  role_id = sdm_role.engineering.id
+}
+```
+
+### Dynamic Access Rules
+
+When using Access Rules, the best practice is to grant Resources access based on
```

Comment:
This sounds like we are giving Resources some kind of access. Maybe we could reword to make it clear that we're giving Roles access to Resources based on type and tags.